ReactAndroid/src/main/java/com/facebook/react/modules/statusbar/StatusBarModule.java

```diff
@@ -201,7 +201,11 @@ public void run() {
                 insetsController.setSystemBarsAppearance(
                     0, WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS);
               }
-            } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            }
+            // Although systemUiVisibility is deprecated in API 30, insetsController doesn't work
+            // reliably and both mechanisms can affect the style. It works properly in API 31, so
+            // use both mechanisms on API 30 only.
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && Build.VERSION.SDK_INT <= Build.VERSION_CODES.R) {
```

Comment:
Can we just do `Build.VERSION.SDK_INT > Build.VERSION_CODES.R` instead? Why use both ways at the same time?

Reply:
Good question. I wasn't confident that even though the new way didn't appear to work on API 30 for me that it still wasn't doing something.

Changing to `>` would be cleaner and I can do that if you prefer.

Reply:
I thought about it and went ahead and updated it to just `>`.